src/main.js

```diff
@@ -18,154 +20,307 @@ import {
+  }
+  // Else, use automated approach to create changeset file
+  else {
+    console.log(
+      "GitHub App is installed and not suspended in the forked repository.\nProceding fwith checks in changelog PR description and automatic creation of changeset file."
+    );
+    await handleAutomaticChangesetCreation(octokit, prData);
+  }
+};
+
+run();
 
-  const changesetFilePath = (prNumber) => `${CHANGESET_PATH}/${prNumber}.yml`;
-  let baseOwner,
-    baseRepo,
-    baseBranch,
-    headOwner,
-    headRepo,
-    headBranch,
-    prNumber,
-    prDescription,
-    prLink;
+// ****************************************************************************
+// II) HELPERS
+// ****************************************************************************
 
+// ----------------------------------------------------------
+// Chnageset Creation Helpers Functions
+// ----------------------------------------------------------
+const handleAutomaticChangesetCreation = async (octokit, prData) => {
+  const changesetCreationMode = "automatic";
   try {
-    // Step 0 - Extract information from the payload
-    ({
-      baseOwner,
-      baseRepo,
-      baseBranch,
-      headOwner,
-      headRepo,
-      headBranch,
-      prNumber,
-      prDescription,
-      prLink,
-    } = extractPullRequestData());
-
-    // Step 1 - Parse changelog entries and validate
     const changelogEntries = extractChangelogEntries(
-      prDescription,
-      processChangelogLine
+      prData.prDescription,
+      processChangelogLine,
+      changesetCreationMode
     );
     const changesetEntriesMap = getChangesetEntriesMap(
       changelogEntries,
-      prNumber,
-      prLink
+      prData.prNumber,
+      prData.prLink,
+      changesetCreationMode
     );
-
-    // Step 2 - Handle "skip" option
-
     if (isSkipEntry(changesetEntriesMap)) {
-      await labelServices.addLabel(
-        octokit,
-        baseOwner,
-        baseRepo,
-        prNumber,
-        SKIP_LABEL
+      await handleSkipEntry(octokit, prData, changesetCreationMode);
+      return;
+    }
+    const changesetFileContent = getChangesetFileContent(changesetEntriesMap);
+    const commitMessage = `Changeset file for PR #${prData.prNumber} created/updated`;
+    await forkedFileServices.createOrUpdateFileInForkedRepoByPath(
+      prData.headOwner,
+      prData.headRepo,
+      prData.headBranch,
+      getChangesetFilePath(prData.prNumber),
+      changesetFileContent,
+      commitMessage
+    );
+    handleLabels(octokit, prData, "remove-all-labels");
+  } catch (error) {
+    await handleErrorChangelogEntries(error, octokit, prData);
+    throw new Error("Automatic creation of changeset file failed.", error);
+  }
+};
+
+const handleManualChangesetCreation = async (octokit, prData) => {
+  // Post info message about adding changeset file manually if PR opened or reopened
+  const changesetCreationMode = "manual";
+  if (prData.prAction == "opened" || prData.prAction == "reopened") {
```

Comment:
The order of operations in my head for this was:

```js
if {isSkipEntry(changesetEntriesMap)} return;
if {changesetFileExist} return

// If neither are true then lets post a message
if (prData.prAction == "opened") {
  await postInfoMessageAboutGitHubAppAndAutomationProcess()
} else {
 await postErrorMessageAboutMissingChangesetFile()
}
```

The reason i suggest this flow is because of 2 edgecases:
1. Unlikely but possible. A user might know beforehand the next PR number and create a changeset file. PR's are incremental, so knowing the next number is just a matter of lookign at the existing PR's/Issues and guessing.
2. Most liekly, an author might add skip to the PR description. In tat case we dont want to show the info

Also why is the info shown on reopen? On opening the PR, there should already be a message that shows this right? Nothing wrong in showing it again, but just wanted to make sure i wasnt missing something

Reply:
Just finished a major refactoring. This includes your edge cases mentioned above.

Also removes the case for a 'reopen' PR action